scenes/game_logic/sequence_puzzle.gd

```diff
@@ -1,9 +1,17 @@
 # SPDX-FileCopyrightText: The Threadbare Authors
 # SPDX-License-Identifier: MPL-2.0
-class_name SequencePuzzle
-extends Node2D
+class_name SequencePuzzle ##nombre global del scrip
+extends Node2D ##se hereda node2d
 
-signal solved
+signal solved ##declara la señal al completar el pussle pero no lo envia(esto se usara despues)
+
+
+signal step_solved(step_index: int) ##señal cuando se complete un step
```

Comment:
This would be an interesting & useful addition to the sequence puzzle code – see discussion on https://github.com/endlessm/threadbare/discussions/748 – but it should not be made as part of this pull request: it should be submitted as a separate, self-contained pull request.

If you need this logic in your StoryQuest, and do not want to split it out for the time being, please duplicate the `sequence_puzzle.gd` script (with a different `class_name`) into your StoryQuest directory, and make the changes you need.